02_bubble/mohitnag/main.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, bubble([]int{3, 2, 1, 5}))
+}
+
+func bubble(s []int) []int {
+	for j := 0; j < len(s); j++ {
+		for i := 1; i < len(s); i++ {
+			if s[i-1] > s[i] {
+				swap(s, i-1, i)
+			}
+		}
```

Comment:
nice work, can add an extra optimisation here as well, if we get to this point and we did not swap any elements then we safely say the array is sorted. You can return early by adding a swapped flag

```go
for j := 0; j < len(s); j++ {
  swapped := false
  for i := 1; i < len(s); i++ {
    if s[i-1] > s[i] {
      swap(s, i-1, i)
      swapped = true
    }
  }
  if !swapped {
    return s
  }
}
```

Reply:
Done